### PR TITLE
Override id field when specified in junit

### PR DESCRIPTION
**Pull request by gnalh**
* Override id field when specified in junit
* Migrate to quick-unit 
* migrate to our junit parser

### CODE REVIEW

File: cli/src/runner.rs [hunk 1]
```diff
@@ -125,29 +126,49 @@ pub async fn extract_failed_tests(
...
+                                id,
+                                org_slug,
+                                repo,
+                            );
+
+                            failures.push(test);
+                        }
                     }
                 }
             }
```
Comment (dfrankland) [inline cli/src/runner.rs:174]: There's a little more copying than necessary here. We can iterate by reference and clone only what's needed. Also, I sprinkled in some small changes to make control flow less verbose.

```suggestion
            for report in junitxml.reports() {
                for suite in &report.test_suites {
                    let parent_name = String::from(suite.name.as_str());
                    for case in &suite.test_cases {
                        if !matches!(case.status, TestCaseStatus::NonSuccess { .. }) {
                            continue;
                        }

                        let name = String::from(case.name.as_str());
                        let classname = case.classname.as_ref().map(|s| String::from(s.as_str()));
                        let file = case.extra.get("file").map(|s| String::from(s.as_str()));
                        let id = case.extra.get("id").map(|s| String::from(s.as_str()));
                        let test = Test::new(
                            name,
                            parent_name.clone(),
                            classname,
                            file,
                            id,
                            org_slug,
                            repo,
                        );

                        failures.push(test);
                    }
                }
            }
```

Also, if wanted, we can be fancy and do something like this too:
```rs
let xml_string_to_string = |s: &quick_junit::XmlString| String::from(s.as_str());
let classname = case.classname.as_ref().map(xml_string_to_string);
let file = case.extra.get("file").map(xml_string_to_string);
let id = case.extra.get("id").map(xml_string_to_string);
```

**Review by dfrankland — APPROVED** [review]

File: context/src/junit/parser.rs [hunk 2]
```diff
@@ -326,6 +331,10 @@ impl JunitParser {
             test_case.extra.insert("filepath".into(), filepath.into());
         }
 
+        if let Some(id) = parse_attr::id(e) {
+            test_case.extra.insert("id".into(), id.into());
```
[the remark this answers is not in the document]
Comment (gnalh) [reply context/src/junit/parser.rs:335]: Yeah I can change that. We aren't doing it above for the other extra attrs. I'll update all of them to match.

[the remark this answers is not in the document]
Comment (dfrankland) [reply context/src/junit/parser.rs:335]: That was probably my mistake. Thanks for cleaning things up!

File: cli/src/runner.rs (hunk 1 above)
Comment (gnalh) [reply cli/src/runner.rs]: I like your style.

File: context/src/junit/validator.rs [hunk 3]
```diff
@@ -65,6 +65,17 @@ pub fn validate(report: &Report) -> JunitReportValidation {
...
+            let test_case_id = uuid::Uuid::parse_str(raw_test_suite_id).unwrap_or_default();
+            if test_case_id.get_version() != Some(uuid::Version::Sha1) {
+                test_suite_validation.add_issue(JunitValidationIssue::Invalid(
+                    JunitTestSuiteValidationIssueInvalid::TestSuiteInvalidId(
+                        raw_test_suite_id.to_string().clone(),
+                    ),
+                ));
+            }
+        }
+
```
Comment (gnalh) [inline context/src/junit/validator.rs:78]: @dfrankland I added some validation + tests. Let me know if you want to review this before I merge.

**Pull request merged**